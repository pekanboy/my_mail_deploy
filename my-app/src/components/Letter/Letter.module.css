.checkbox {
    min-width: 20px;
    min-height: 20px;

    background: #FFFFFF;
    border: 1px solid #000000;
    box-sizing: border-box;
    border-radius: 4px;
}

.avatar {
    width: 30px;
    height: 30px;

    border-radius: 4px;
    margin-right: 8px;
    margin-left: 8px;
}

.name {
    max-width: 140px;
    width: 140px;

    font-family: "Roboto Thin", sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 17px;
    align-items: center;
    text-align: left;

    color: #000000;

    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;

    margin-right: 10px;
}

.theme {
    max-width: 15%;

    font-family: "Roboto Light", sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 30px;
    align-items: center;
    text-align: left;

    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;

    margin-right: 10px;
}

.snippet {
    max-width: 600px;
    width: 65%;

    font-family: "Roboto Light", sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 17px;
    align-items: center;
    text-align: left;

    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;

    opacity: 50%;
}

.container {
    display: flex;
    align-items: center;

    max-width: 1040px;
    height: 48px;

    background: white;
    border: 1px solid rgba(217, 220, 224, 0.86);

    position: relative;

    padding: 0 8px;
}

.container:hover {
    background: #E5E5E5;
}

.selected {
    transition-duration: 1s;
    background-color: #ebecef;
    opacity: 92%;
}
