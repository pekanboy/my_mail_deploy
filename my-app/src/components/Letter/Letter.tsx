import React from 'react';
import styles from './Letter.module.css';
import {ILetter} from '../../interfaces/ILetter';

export interface LetterProps {
    letter: ILetter
    checkHandler: (id: number) => void
}

export const Letter: React.FC<LetterProps> = ({checkHandler, letter}) => {
    return (
        <div
            className={`${styles.container} 
                ${letter.isSelected && styles.selected}`}
            id={letter.id.toString()}>
            <input
                className={styles.checkbox}
                type="checkbox"
                onChange={() => {
                    checkHandler(letter.id);
                }
                }
                checked={!!letter.isSelected}/>
            <img
                className={styles.avatar}
                src={letter.avatar}
                alt="Аватар пользователя"/>
            <span className={styles.name}>{letter.sender}</span>
            <span className={styles.theme}>{letter.theme}</span>
            <span className={styles.snippet}>{letter.snippet}</span>
        </div>
    );
};
