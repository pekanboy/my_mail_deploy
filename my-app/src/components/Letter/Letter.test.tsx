import React from 'react';
import {configure, shallow, ShallowWrapper} from 'enzyme';
import {Letter, LetterProps} from './Letter';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const testData: LetterProps = {
    checkHandler: () => undefined,
    letter: {
        'id': 1,
        'avatar': 'http://i.pravatar.cc/300?u=1',
        'sender': 'Бединге',
        'theme': 'Carolina Hurricanes',
        'snippet': 'Люди всё время меня спрашивают: знаю ли я Тайлера Дёрдена?',
        'isSelected': false,
    },
};


describe('Test Letter component', () => {
    it('to match snapshot ', function() {
        const wrapper: ShallowWrapper = shallow(<Letter {...testData} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('test callback called', () => {
        const mockClick = jest.fn();
        const data: LetterProps = {...testData, checkHandler: mockClick};

        const wrapper: ShallowWrapper = shallow(<Letter {...data} />);

        wrapper.find('input[type="checkbox"]').simulate('change', {});

        expect(mockClick.mock.calls.length).toBe(1);
    });

    it('test selected letter', () => {
        const data = testData;
        data.letter = {...data.letter, isSelected: true};
        const wrapper: ShallowWrapper = shallow(<Letter {...data} />);

        expect(wrapper).toMatchSnapshot();
    });
});
