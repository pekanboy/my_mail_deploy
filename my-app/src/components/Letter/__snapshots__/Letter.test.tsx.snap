// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Letter component test selected letter 1`] = `
<div
  className="container 
                selected"
  id="1"
>
  <input
    checked={true}
    className="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <img
    alt="Аватар пользователя"
    className="avatar"
    src="http://i.pravatar.cc/300?u=1"
  />
  <span
    className="name"
  >
    Бединге
  </span>
  <span
    className="theme"
  >
    Carolina Hurricanes
  </span>
  <span
    className="snippet"
  >
    Люди всё время меня спрашивают: знаю ли я Тайлера Дёрдена?
  </span>
</div>
`;

exports[`Test Letter component to match snapshot  1`] = `
<div
  className="container 
                false"
  id="1"
>
  <input
    checked={false}
    className="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <img
    alt="Аватар пользователя"
    className="avatar"
    src="http://i.pravatar.cc/300?u=1"
  />
  <span
    className="name"
  >
    Бединге
  </span>
  <span
    className="theme"
  >
    Carolina Hurricanes
  </span>
  <span
    className="snippet"
  >
    Люди всё время меня спрашивают: знаю ли я Тайлера Дёрдена?
  </span>
</div>
`;
